lib/Runtime/Library/JavascriptBigInt.cpp

```diff
@@ -113,26 +156,182 @@ namespace Js
         }
     }
 
-    bool JavascriptBigInt::MulThenAdd(uint32 digitMul, uint32 digitAdd)
+    // return low(a+b) and out carry
+    digit_t JavascriptBigInt::AddDigit(digit_t a, digit_t b, digit_t* carry)
+    {
+        digit_t result = a + b;
+        if (result < a)
+        {
+            *carry += 1;
+        }
+        return result;
+    }
+
+    // return low(a+b) and out carry
+    digit_t JavascriptBigInt::SubDigit(digit_t a, digit_t b, digit_t* borrow)
+    {
+        digit_t result = a - b;
+        if (result > a)
+        {
+            *borrow += 1;
+        }
+        return result;
+    }
+
+    bool JavascriptBigInt::IsZero(JavascriptBigInt * pbi)
+    {
+        return (pbi->m_length == 1 && pbi->m_digits[0] == 0);
+    }
+
+    JavascriptBigInt * JavascriptBigInt::AbsoluteIncrement(JavascriptBigInt * pbi)
+    {
+        JavascriptBigInt* result = pbi;
```

Comment:
Should we make a new one at this point? I expect that in-place changes might end up being useful as a performance optimization, but BigInt is meant to be logically immutable. Consider the following:

```c++
x = 123n;
y = x;
++x;
print(y); // still 123
```

Reply:
fixed